python/lsst/ap/verify/pipeline_driver.py

```diff
@@ -66,98 +46,50 @@ class ApPipeParser(argparse.ArgumentParser):
-    files shall be added to `job`. This method is a temporary workaround
-    for the `verify` framework's limited persistence support, and will be
-    removed in a future version.
-
-    Parameters
-    ----------
-    metadata : `lsst.daf.base.PropertySet`
-        The full metadata from running a task(s). Assumed to contain keys of
-        the form "<standard task prefix>.verify_json_path" that maps to the
-        absolute file location of that task's serialized measurements.
-        All other metadata fields are ignored.
-    job : `lsst.verify.Job`
-        The Job object to which to add measurements. This object shall be
-        left in a consistent state if this method raises exceptions.
-
-    Raises
-    ------
-    lsst.ap.verify.pipeline_driver.MeasurementStorageError
-        Raised if a "verify_json_path" key does not map to a string, or serialized
-        measurements could not be located or read from disk.
-    """
-    try:
-        keys = metadata.names(topLevelOnly=False)
-        files = [metadata.getAsString(key) for key in keys if key.endswith('verify_json_path')]
-
-        for measurementFile in files:
-            with open(measurementFile) as f:
-                taskJob = Job.deserialize(**json.load(f))
-            job += taskJob
-    except (IOError, TypeError) as e:
-        raise MeasurementStorageError('Task metadata could not be read; possible downstream bug') from e
-
-
-def runApPipe(metricsJob, workspace, parsedCmdLine):
+def runApPipe(workspace, parsedCmdLine):
     """Run `ap_pipe` on this object's dataset.
 
     Parameters
     ----------
-    metricsJob : `lsst.verify.Job`
-        The Job object to which to add any metric measurements made.
     workspace : `lsst.ap.verify.workspace.Workspace`
         The abstract location containing input and output repositories.
     parsedCmdLine : `argparse.Namespace`
         Command-line arguments, including all arguments supported by `ApPipeParser`.
 
-    Raises
-    ------
-    lsst.ap.verify.pipeline_driver.MeasurementStorageError
-        Raised if measurements were made, but `metricsJob` could not be updated
-        with all of them. This exception may suppress exceptions raised by
-        the pipeline itself.
+    Returns
+    -------
+    dataIds : `lsst.pipe.base.DataIdContainer`
+        The set of complete data IDs fed into ``ap_pipe``.
     """
     log = lsst.log.Log.getLogger('ap.verify.pipeline_driver.runApPipe')
 
-    dataId = _parseDataId(parsedCmdLine.dataId)
+    args = [workspace.dataRepo,
+            "--output", workspace.outputRepo,
+            "--calib", workspace.calibRepo,
+            "--template", workspace.templateRepo]
+    args.extend(_getConfigArguments(workspace))
+    if parsedCmdLine.dataId:
+        args.extend(["--id", *parsedCmdLine.dataId.split(" ")])
+    else:
+        args.extend(["--id"])
+    args.extend(["--processes", str(parsedCmdLine.processes)])
```

Comment:
Hooray for forward-thinking optimism 💯

Reply:
It supports multiprocessing as of this ticket...? 😕